### ISSUE TITLE
DocumentDB E3030 '8.0.0' is not one of ['3.6.0', '4.0', '4.0.0', '5.0.0']

### ISSUE THREAD

**Issue by nuwanwre** [aws-cloudformation/cfn-lint#4389]
### CloudFormation Lint Version

1.46.0

### What operating system are you using?

Ubuntu

### Describe the bug

DocumentDB 8.0 is generally available since end of last year [1].
`cfn-lint` throws a `E3030` error mentioning `8.0.0` is not supported. 

[1] https://docs.aws.amazon.com/documentdb/latest/developerguide/release-notes.html#release-notes.11-13-2025

### Expected behavior

Also allow version 8.0.0

### Reproduction template

```yaml
DocumentDBCluster:
    Type: "AWS::DocDB::DBCluster"
    Properties:
      DBClusterIdentifier: "docdb-80-cluster"
      DBSubnetGroupName: !Ref DatabaseSubnetGroup
      MasterUsername: ""
      MasterUserPassword: ""
      Port: 27017
      VpcSecurityGroupIds:
        - !Ref DatabaseSecurityGroup
      EngineVersion: "8.0.0"
```
